p3/plot/_cascade.py

```diff
@@ -301,133 +84,12 @@ def cascade(df, eff=None, size=(6, 5), **kwargs):
 
     if len(df["problem"].unique()) > 1:
         raise NotImplementedError(
-            "Handling multiple problems is currently not implemented."
-        )
-
-    kwargs.setdefault("platform_legend", Legend())
-    plat_legend = kwargs["platform_legend"]
-    plat_legend.kwargs.setdefault("ncols", 4)
-    plat_legend.kwargs.setdefault("loc", "upper center")
-    plat_legend.kwargs.setdefault("bbox_to_anchor", (0.5, 0.0))
-
-    kwargs.setdefault("application_legend", Legend())
-    app_legend = kwargs["application_legend"]
-
-    kwargs.setdefault("platform_style", PlatformStyle())
-    plat_style = kwargs["platform_style"]
-    if not plat_style.colors:
-        plat_style.colors = getattr(plt.cm, "RdBu")
-
-    kwargs.setdefault("application_style", ApplicationStyle())
-    app_style = kwargs["application_style"]
-    if not app_style.colors:
-        app_style.colors = getattr(plt.cm, "tab10")
-    if not app_style.markers:
-        app_style.markers = getattr(
-            matplotlib.markers.MarkerStyle, "filled_markers"
+            "Handling multiple problems is currently not implemented.",
         )
 
-    # Choose the efficiency column based on eff parameter and available columns
-    efficiency_columns = []
-    for column in ["app eff", "arch eff"]:
-        if column in df:
-            efficiency_columns.append(column)
-    if len(efficiency_columns) == 0:
-        msg = "DataFrame does not contain an 'app eff' or 'arch eff' column."
-        raise ValueError(msg)
-
-    if eff is None:
-        eff_column = efficiency_columns[0]
+    kwargs.setdefault("backend", "matplotlib")
+    backend = kwargs["backend"]
+    if backend == "matplotlib":
+        return MPLCascadePlot(df, eff, size, **kwargs)
```

Comment:
We can defer the import until we need it like this:

```suggestion
        from p3.plot.backend.matplotlib import CascadePlot
        return CascadePlot(df, eff, size, **kwargs)
```

...and that way, there's no dependency on matplotlib unless it's the backend in use.

Reply:
Done.